tests/monotouch-test/CoreData/NSQueryGenerationTokenTest.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+#if XAMCORE_2_0
+using CoreData;
+using Foundation;
+#else
+using MonoTouch.CoreData;
+using MonoTouch.Foundation;
+#endif
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.CoreData
+{
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class NSQueryGenerationTokenTest
+	{
+		[Test]
+		public void EncodeWithCoderTest ()
+		{
+			// Added test to ensure we do support NSCoding even when introspection fails.
+			if (IntPtr.Size == 8  && TestRuntime.CheckExactXcodeVersion (9, 0, beta: 1)) {
```

Comment:
Why did you add an exact check for Xcode 9 b1?

In your case it means this test won't run on any other Xcode version (usually exact Xcode checks like this are used to _skip_ a test for a specific Xcode version, so that the test runs in later Xcode versions, which allows us to manually verify if an SDK bug was fixed (or not)).